Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -686,10 +704,18 @@ def virtual_service_create_command(client: PolicyComputeEngine, args: Dict[str,
     proto = convert_protocol(protocol)
 
     service = VirtualService(name=name, service_ports=[ServicePort(port=port, proto=proto)])  # type: ignore
-    virtual_service = client.virtual_services.create(service)  # type: ignore
-
-    virtual_service_json = virtual_service.to_json()
-    readable_output = prepare_virtual_service_output(virtual_service_json)
+    try:
+        virtual_service = client.virtual_services.create(service)  # type: ignore
+        virtual_service_json = virtual_service.to_json()
+        virtual_service_json["already_exists"] = False
+        readable_output = prepare_virtual_service_output(virtual_service_json)
+    except IllumioException:
+        virtual_services = client.virtual_services.get(params={"name": name})
+        for virtual_service in virtual_services:
+            virtual_service_json = virtual_service.to_json()
```

Comment:
The response values are objects. There's no need to convert them to a dict before the name check.

Something along the lines of

```
...
for virtual_service in virtual_services:
    if virtual_service.name == name:
        readable_output = prepare_virtual_service_output(virtual_service.to_json())
```